src/main/java/emu/grasscutter/game/ability/HSDAbilityManager.java

```diff
@@ -178,11 +181,18 @@ public void checkAndHeal(GameEntity sourceEntity, String modifierString) {
                         curAttack = healActionAvatar.getFightProperty(FightProperty.FIGHT_PROP_CUR_ATTACK);
                         curDefense = healActionAvatar.getFightProperty(FightProperty.FIGHT_PROP_CUR_DEFENSE);
 
-                        //Special case for Hu Tao:
+                        //Special case for Q skill of Hu Tao
                         if(healDataAvatar.avatarName.equals("Hutao") && curHP <= maxHP * 0.5 && ratio != 0) {
                             ratio = 0.1555f;
                         }
 
+                        //Special case for E skill of Hu Tao
+                        if(healDataAvatar.avatarName.equals("Hutao") && skillId == 10462) {
```

Comment:
instead of hardcoding a system for *specifically* Hu Tao, consider adding a *system* for this